drivers/adc/adc_nrfx_saadc.c

```diff
@@ -5,6 +5,8 @@
  */
 
 #define ADC_CONTEXT_USES_KERNEL_TIMER
+#define ADC_DRIVER_USES_INTERNAL_TIMER
```

Comment:
Given this is always defined, why define it? just remove this definition and all the ifdefs accompanying it :)